new_ui/app.py

```
@@ -143,6 +145,110 @@ async def get_styles():
+                # No new yields — send heartbeat every ~5s of idle
+                idle_count += 1
+                if idle_count >= 50:  # 50 * 100ms = 5s
+                    await websocket.send_json({"type": "heartbeat"})
+                    update_heartbeat()
+                    idle_count = 0
+
+            await asyncio.sleep(0.1)
+
+    except WebSocketDisconnect:
+        logger.debug("WebSocket client disconnected")
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)